ironic-deployment/overlays/e2e-release-24.0-with-inspector/kustomization.yaml

```diff
@@ -26,12 +26,16 @@ images:
 secretGenerator:
 - name: ironic-htpasswd
   behavior: create
-  envs:
-  - ironic-htpasswd
+  files:
+  - username=ironic-username
+  - password=ironic-password
```

Comment:
The secret is called `ironic-htpasswd` because it used to contain ah [htpasswd formatted file](https://httpd.apache.org/docs/2.4/programs/htpasswd.html). Now it just contains the username and password as plain text. If we really intend to have them as plain text we should change the name to avoid confusion.

Reply:
Didn't it just contain an htpasswd string generated by the command? Basically ironic-image will now use the same command to generate the htpsasswd itself, so it needs the ironic username and password as input. Since it will eventually be used to generate the same htpasswd, I think keeping the name the same makes sense.

Reply:
Clarified offline, commenting for closure.
The issue is that we then give ironic access to the password in plain text, which is more information than it needs. The hash is enough for its purpose so it is better to "limit" the access level to that :slightly_smiling_face:

Reply:
See above. This is fake security: if you have read access to the Ironic pod, you have all BMC credentials and all configdrives. In that cases, knowing the API password is not overly useful anymore.

Reply:
Agreed. @MahnoorAsghar please change the PR so it is clear we are going to *change* the ironic credentials to be plain text *and* we are going to mount the secrets instead of consuming them as environment variables. My confusion came from the description that we are going to mount *htpasswd* as a volume which we do not.